source/game/system/map/MapdataCheckPoint.cc

```diff
@@ -24,11 +25,113 @@ void MapdataCheckPoint::read(EGG::Stream &stream) {
     m_nextPt = stream.read_u8();
 }
 
+/// @brief Calculates @ref m_nextPoints and @ref m_prevPoints from @ref m_nextPt and @ref m_prevPt.
+/// @details Also calculates the quadrilaterals for the next checkpoints, filling the fields of @ref
+/// LinkedCheckpoint for each.
+/// @addr{0x80515624}
+void MapdataCheckPoint::initCheckpointLinks(MapdataCheckPointAccessor &accessor, int id) {
+    m_id = id;
+    auto courseMap = CourseMap::Instance();
+
+    // Calculate the quadrilateral's `m_prevPoints`. If the check point is the first in its group
+    // (prev == -1), it has multiple previous checkpoints defined by its preceding checkpaths
+    if (m_prevPt == 0xff) {
```

Comment:
Shouldn't these be `u8` instead of `s8` since in your comment makes it sound like they are interpreted as -1